packages/lodestar/src/api/rest/routes/validator/produceAttestation.ts

```diff
@@ -0,0 +1,60 @@
+import {IFastifyServer} from "../../index";
+import fastify, {DefaultQuery} from "fastify";
+import {IApiModules} from "../../../interface";
+import {IncomingMessage, Server, ServerResponse} from "http";
+import {toRestJson} from "../../utils";
+import {produceAttestation} from "../../../impl/validator";
+
+interface Query extends DefaultQuery {
+  validator_pubkey: string;
+  poc_bit: number;
+  slot: number;
+  shard: number;
+}
+
+
+const opts: fastify.RouteShorthandOptions<Server, IncomingMessage, ServerResponse, Query> = {
+  schema: {
+    querystring: {
+      type: 'object',
+      required: ["validator_pubkey", "poc_bit", "slot", "shard"],
+      properties: {
+        "randvalidator_pubkeyao_reveal": {
+          type: "string"
+        },
+        "poc_bit": {
+          type: "integer",
+          minimum: 0
+        },
+        shard: {
+          type: "integer",
+          minimum: 0
+        },
+        slot: {
+          type: "integer",
+          minimum: 0
+        },
+      }
+    },
+  }
+};
+
+export const registerAttestationProductionEndpoint = (fastify: IFastifyServer, modules: IApiModules): void => {
```

Comment:
I think you should use a ValidatorApi instance at this level instead of exposing/passing around the underlying modules.
The idea is that we can/should used the same exact implementation, the Validator/BeaconApi modules, which (should) provide the exact interface needed for the outside world. Then the Validator/BeaconApi can be the only places needed to maintain the underlying implementation.

This is roughly the sketch I was thinking:
BeaconNode
  - instantiate core modules
  - instantiate api modules using core modules (publicly accessible properties)
  - instantiate rpc/rest api modules using api modules

Then in all cases, we're using the exact same implementations. 
- within a js application, `node.apis.validator.produceAttestation(...)`
- a bundled validator, `new Validator(new RpcBlah(node.apis.validator))`
- external api, `/attestation` ends up hitting `node.apis.validator.produceAttestation`

Reply:
I was going to do the same but realized we will shoot ourself in foot in the near future as implementations for rpc and http will most likely diverged. First issue will be soon added event streaming where we would have to subscribe to module events in http while that isn't possible in rpc. 

My idea here, although not fully realized, is to extract implementation logic inside /api/impl and use it from both rpc and http "controllers". That way http/rpc "controllers" can handle protocol specific quirks like serialization/deserialization while using same easily testable logic.

Other point is that if user doesn't wan't rpc enabled I would have to create JsonRpc as it initializes api.

Reply:
Should I extract api classes from rpc and in rpc create some wrapper around them?

Reply:
I wonder if we should even support json-rpc anymore

Reply:
we have to because you cannot have rest api in browser. That said light clients could/should inject library anyways

Reply:
browser can't support a json-rpc server either ;)
it can be a client of either tho

Reply:
it can via web rtc, json-rpc is transport agnostic